### PR TITLE
[Assignment #2] Refactor out version to plugin

### PR DESCRIPTION
**Pull request by cwilgenhoff**
Requesting CR.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -2,35 +2,31 @@
...
-                return reply({ version: Package.version });
-            }
-        }
-    });
-
-    server.start(function (err) {
+    server.connection(internals.connection);
 
+    server.register(Version, function (err) {
         Hoek.assert(!err, err);
```
Comment (chyld) [inline lib/index.js]: newline after function

File: lib/index.js [hunk 2]
```diff
@@ -2,35 +2,31 @@
...
-        console.log('Server started at: ' + server.info.uri);
+    
+        return server.start(function (err) {
```
Comment (chyld) [inline lib/index.js]: why are you using return?

Comment (cwilgenhoff) [inline lib/index.js]: My bad.

File: lib/version.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        config: {
+            description: 'Returns the version of the server',
+            handler: function (request, reply) {
+
+                return reply(internals.response);
+            }
+        }
+    });
+    
+    next();
```
Comment (AdriVanHoudt) [inline lib/version.js]: use `return next()` for readability

File: lib/version.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+exports.register.attributes = {
+    name: 'version'
+};
```
Comment (AdriVanHoudt) [inline lib/version.js]: newline at the end of the file

File: package.json [hunk 5]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "hueniversity",
-  "version": "0.0.1",
+  "version": "0.0.2",
   "description": "Community learning experiment",
-  "main": "lib/index.js",
```
Comment (AdriVanHoudt) [inline package.json]: why change this?

File: lib/version.js (hunk 3 above)
Comment (TheAlphaNerd) [inline lib/version.js]: fwiw I believe the return serves a pragmatic purpose beyond readability.  It also ensures that the context / stack frame of this function is appropriately exited when the callback is executed.  This ensures that no weird behavior happens such as unexpected data manipulation, race conditions, or even a second callback call (EEP)

Comment (AdriVanHoudt) [inline lib/version.js]: you are right it also prevents calling callbacks twice and stuff, although in this particular case that wouldn't happen but I completely agree, I use it where I can, in this case even in the init function in index.js

File: package.json (hunk 5 above)
Comment (cwilgenhoff) [inline package.json]: This is due I've added an index file into the base project folder, so no need to go inside lib to run it.

Comment (AdriVanHoudt) [inline package.json]: that behaviour was specifically removed from solution of the first assignment and is not included in the second one so no need to change it

File: lib/version.js (hunk 3 above)
Comment (cwilgenhoff) [inline lib/version.js]: I don't understand the need of a return, as you said, in this particular case callbacks are not getting called more than once at a time.

File: package.json (hunk 5 above)
Comment (cwilgenhoff) [inline package.json]: Oh, sorry my bad for not having started since assignment 1. Thanks for the heads up.

Comment (AdriVanHoudt) [inline package.json]: Np, try to only modify what is in the assignment and not add extra stuff (like this or e.g. tests or something) when not asked ;)

File: lib/version.js (hunk 3 above)
Comment (AdriVanHoudt) [inline lib/version.js]: In this case readability, when quickly reading the code I can immediately see where the callback is called take a look at https://gist.github.com/hueniverse/a06f6315ea736ed1b46d#return-callbacks

File: lib/index.js [hunk 6]
```diff
@@ -2,35 +2,31 @@
...
-var Package = require('../package.json');
+var Version = require('./version.js');
 
 
 // Declare internals
 
-var internals = {};
+var internals = {
+    connection: {
+        port: process.env.PORT || 8000
```
Comment (hueniverse) [inline lib/index.js]: Adding env support wasn't part of the assignment.

Comment (cwilgenhoff) [inline lib/index.js]: Sorry, I was developing on c9, that was the reason to use it. Removed.

**Pull request closed without merging**